app/controllers/character_weapons_controller.rb

```diff
@@ -0,0 +1,68 @@
+class CharacterWeaponsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_character_weapon, except: [:new, :create]
+  before_action :verify_character_weapon, except: [:new, :create]
+
+  def destroy
+    @character_weapon.destroy
+    redirect_to @character_weapon.character, flash: { notice: "Weapon #{@character_weapon.weapon.name} has been destroyed" }
+  end
+
+  def show
+    @weapon = @character_weapon.weapon
+  end
+
+  def new
+    @weapons = weapon_list
+    @character = current_character
+  end
+
+  def create
+    @character_weapon = CharacterWeapon.new
+    @character_weapon.new_basic_weapon(current_character, current_weapon)
```

Comment:
I don't feel this method should be a instance method, it should be at the class scope and return the instance o something like that